include/gridtools/stencil_composition/backend_naive/iterate_domain_naive.hpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <type_traits>
+
+#include "../../common/defs.hpp"
+#include "../../common/generic_metafunctions/for_each.hpp"
+#include "../../common/hymap.hpp"
+#include "../../meta.hpp"
+#include "../../storage/common/storage_info.hpp"
+#include "../dim.hpp"
+#include "../grid.hpp"
+#include "../iterate_domain_fwd.hpp"
+#include "../local_domain.hpp"
+#include "../pos3.hpp"
+#include "../sid/concept.hpp"
+#include "../sid/multi_shift.hpp"
+
+namespace gridtools {
+    /**
+     * @brief iterate domain class for the naive backend
+     */
+    template <class LocalDomain>
+    class iterate_domain_naive {
+        GT_STATIC_ASSERT(is_local_domain<LocalDomain>::value, GT_INTERNAL_ERROR);
+
+        using ptr_map_t = typename LocalDomain::ptr_map_t;
+        using strides_map_t = typename LocalDomain::strides_map_t;
+
+        template <class Dim>
+        struct increment_f {
+            iterate_domain_naive *m_self;
+            int_t m_offset;
+
+            template <class Arg>
+            void operator()() const {
+                auto const &stride = sid::get_stride<Dim>(m_self->template strides<Arg>());
+                auto &ptr = at_key<Arg>(m_self->m_ptr_map);
+                sid::shift(ptr, stride, m_offset);
+            }
+        };
+
+        template <class Grid>
+        struct set_base_offset_f {
```

Comment:
Why is this base offset backend specific?

Reply:
It is not. This code is moved here temporary and eventually it will gone . The perpose is to make naive backend as much stand alone as possible.